pipeline/metadata/test_flatten.py

```diff
@@ -1109,9 +1156,58 @@ def test_flattenmeasurement_dns(self) -> None:
         'error': None,
         'anomaly': True,
         'success': True,
+        'controls_failed': False,
         'received': {'ip': '104.20.161.134', 'matches_control': ''},
         'rcode': ["0", "0", "0"],
         'measurement_id': ''
+      },
+      {
+        'domain': 'login.live.com',
+        'category': 'Communication Tools',
+        'ip': '8.8.4.4',
+        'date': '2021-03-15',
+        'start_time': '2021-03-15T20:02:06.996014164-04:00',
+        'end_time': '2021-03-15T20:02:07.015386713-04:00',
+        'error': None,
+        'anomaly': None,
+        'success': True,
+        'controls_failed': False,
+        'received': {'ip': '40.126.31.135'},
+        'rcode': ['0', '0', '0'],
+        'measurement_id': '',
+        'has_type_a': True
+      },
+      {
+        'domain': 'login.live.com',
+        'category': 'Communication Tools',
+        'ip': '8.8.4.4',
+        'date': '2021-03-15',
+        'start_time': '2021-03-15T20:02:06.996014164-04:00',
+        'end_time': '2021-03-15T20:02:07.015386713-04:00',
+        'error': None,
+        'anomaly': None,
+        'success': True,
+        'controls_failed': False,
+        'received': {'ip': '40.126.31.8'},
+        'rcode': ['0', '0', '0'],
+        'measurement_id': '',
+        'has_type_a': True
+      },
+      {
+        'domain': 'login.live.com',
+        'category': 'Communication Tools',
+        'ip': '8.8.4.4',
+        'date': '2021-03-15',
+        'start_time': '2021-03-15T20:02:06.996014164-04:00',
+        'end_time': '2021-03-15T20:02:07.015386713-04:00',
+        'error': None,
+        'anomaly': None,
+        'success': True,
+        'controls_failed': False,
+        'received': {'ip': '40.126.31.6'},
+        'rcode': ['0', '0', '0'],
+        'measurement_id': '',
+        'has_type_a': True
```

Comment:
I can't comment on the exact line, but at line 1121 can you change the code

```
for row in rows:
  row['measurement_id'] = ''
  result.append(row.copy())
```

To something like

```
results.append(list(rows))
for result in results:
  result['measurement_id'] = ''
```

I'm worried the .copy() in that code may be hiding a bug where we're yielding the same row object multiple times while overwriting its ip field.

Reply:
Done